qrenderdoc/Windows/ShaderViewer.cpp

```diff
@@ -1486,10 +1500,7 @@ void ShaderViewer::SetTextAndUpdateMargin0(ScintillaEdit *sc, const QString &tex
 
 void ShaderViewer::readonly_keyPressed(QKeyEvent *event)
 {
-  if(event->key() == Qt::Key_F && (event->modifiers() & Qt::ControlModifier))
-  {
-    m_FindReplace->setReplaceMode(false);
-
+  auto SetFindTextFromCurrentWord = [this]() {
```

Comment:
I feel this would be better as a plain member function since there's no particular need for it to be a lambda to capture here.

Reply:
I made it a lambda only because that code is used twice in the same function.
Is it really worth creating a member function for this?

Reply:
I'm not quite sure I'm on the same page - there's no cost to adding a member function so I'm not sure how we'd say it's worth it or not? To me that would be more a question of whether it's worth duplicating the code vs making a function.

I would always reach for a normal function first and go to a lambda if there's a reason it becomes clumsy or awkward, especially since e.g. it's more annoying to step through for debug.

In my view lambdas are good for cases like anonymous callbacks (happens a lot in the UI for invoking between threads), or to grab some local context where it would be clumsy to pass that context in via an additional function parameter. In this case though this is pure code reuse so a member function should be well suited.

Reply:
I mean that since this code is only used inside this function and not meant to be part of the ShaderViewer's "API", I figured it would be better to just create a local lambda.
But if you prefer a member function, I have absolutely no problem with that! :)

Reply:
Ok I made the lambda a member function.
I think it would be nice to merge this PR into the main branch to at least get the functionality.
I'll work on adding bookmark button (and perhaps a list of bookmarks in a dropdown list like you suggest) in another PR.
Does that make sense?

Reply:
I'd rather wait until you're able to make those follow up changes. I don't think there's any rush to get this in so the PR can wait open until you have a chance to add to it and the feature is more well-rounded and complete.